controllers/healthcheck_controller.go

```diff
@@ -268,105 +270,153 @@ func (r *HealthCheckReconciler) createRBACForWorkflow(log logr.Logger, hc *activ
 	}
 
 	if workFlowType != remedy {
-		servacc, err := r.createServiceAccount(r.kubeclient, hcSa, wfNamespace)
+		servacc, created, err := r.createServiceAccount(r.kubeclient, hcSa, wfNamespace)
 		if err != nil {
 			log.Error(err, "Error creating ServiceAccount for the workflow")
 			r.Recorder.Event(hc, v1.EventTypeWarning, "Warning", "Error creating ServiceAccount for the workflow")
 			return err
 		}
-		log.Info("Successfully Created", "ServiceAccount", servacc)
-		r.Recorder.Event(hc, v1.EventTypeNormal, "Normal", "Successfully Created ServiceAccount")
+		if created {
```

Comment:
Moving this logic to the respective functions makes it simple so that these conditions do not have to be repeated each time they are used.

Reply:
moved logs and events to respective function.